code/modules/clothing/under/striketeam.dm

```diff
@@ -0,0 +1,143 @@
+var/list/deathsquad_uniforms = list()
+
+var/list/holomap_cache = list()
+
+/obj/item/clothing/under/deathsquad
+	name = "deathsquad holosuit"
+	desc = "A state-of-the-art suit featuring an holographic map of the area, to help the squad coordinate their efforts."
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/altsuits.dmi', "right_hand" = 'icons/mob/in-hand/right/altsuits.dmi')
+	icon_state = "deathsquad"
+	item_state = "deathsquad"
+	_color = "deathsquad"
+	flags = FPRINT  | ONESIZEFITSALL
+	action_button_name = "Toggle Holomap"
+
+/obj/item/clothing/under/deathsquad/New()
+	..()
+	deathsquad_uniforms += src
+
+/obj/item/clothing/under/deathsquad/Destroy()
+	deathsquad_uniforms -= src
+
+	for(var/cacheIcon in holomap_cache)
+		if(findtext(cacheIcon, "\ref[src]"))
+			var/image/I = holomap_cache[cacheIcon]
+			animate(I)
+			qdel(I)
+			holomap_cache -= cacheIcon
+	..()
+
+/obj/item/clothing/under/deathsquad/ui_action_click()
+	togglemap()
+
+/obj/item/clothing/under/deathsquad/process()
+	update_holomap()
+
+/obj/item/clothing/under/proc/togglemap()
+	if(usr.isUnconscious())
+		return
+
+	if(!ishuman(usr))
+		to_chat(usr, "<span class='warning'>Only humanoids can wear this suit</span>")
+		return
+
+	var/mob/living/carbon/human/H = usr
+
+	if(H.get_item_by_slot(slot_w_uniform) != src)
+		to_chat(H, "<span class='warning'>You need to wear the suit first</span>")
+		return
+
+	if(src.holomap_activated)
+		deactivate_holomap()
+		to_chat(H, "<span class='notice'>You disable the holomap.</span>")
+	else
+		holomap_activated = 1
+		activator = H
+		processing_objects.Add(src)
+		process()
+		to_chat(H, "<span class='notice'>You enable the holomap.</span>")
+
+#define HOLOMAP_ERROR	0
+#define HOLOMAP_YOU		1
+#define HOLOMAP_OTHER	2
+#define HOLOMAP_DEAD	3
+
+/obj/item/clothing/under/proc/update_holomap()
+
+/obj/item/clothing/under/deathsquad/update_holomap()
+	var/turf/T = get_turf(src)
+	if(!T)//nullspace begone!
+		return
+
+	if((activator.get_item_by_slot(slot_w_uniform) != src) || (!activator.client) || (holoMiniMaps[T.z] == null))
+		deactivate_holomap()
+		return
+
+	activator.client.images -= holomap_images
+
+	holomap_images.len = 0
```

Comment:
The considering the images are re-added a few lines bellow I consider we only need to add that line when deactivating the holominimaps.